script.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
```

Comment:
- For `my_each`, `my_each_with_index` and `my_select` methods, handle the case when no block is given, they should return an Enumerator.

This is for the built-in methods
`[1,2,3,4].select
=> #<Enumerator: [1, 2, 3, 4]:select>`

`[1,2,3,4].each
=> #<Enumerator: [1, 2, 3, 4]:each>`

` [1,2,3,4].each_with_index
=> #<Enumerator: [1, 2, 3, 4]:each_with_index>`

-All three methods return the error `LocalJumpError:
       no block given (yield)
`
- The error occurs because your methods are always expecting a block, however, there are cases when a block isn't provided.

Reply:
my_each method updated